base/strings/unicode.jl

```diff
@@ -385,7 +386,14 @@ julia> islowercase('❤')
 false
 ```
 """
-islowercase(c::AbstractChar) = ismalformed(c) ? false : Bool(ccall(:utf8proc_islower, Cint, (UInt32,), UInt32(c)))
+function islowercase(c::AbstractChar)
+    if ismalformed(c)
+        false
+    else
+        ui = UInt32(c)::UInt32 # type-assertion to ensure that we may assume :foldable
+        @assume_effects :foldable Bool(ccall(:utf8proc_islower, Cint, (UInt32,), ui))
+    end
```

Comment:
May be slightly simpler:
```suggestion
    ismalformed(c) && return false
    ui = UInt32(c)::UInt32 # type-assertion to ensure that we may assume :foldable
    return @assume_effects :foldable ccall(:utf8proc_islower, Cint, (UInt32,), ui) != 0
```